checker/manual/creating-a-checker.tex

```diff
@@ -1094,31 +1087,84 @@ \section{The checker class:  Compiler interface\label{writing-compiler-interface
 checker classes for the Interning Checker and the Nullness Checker:
 
 \begin{Verbatim}
-  @TypeQualifiers({ Interned.class, PolyInterned.class, PolyAll.class })
   @SupportedLintOptions({"dotequals"})
   public final class InterningChecker extends BaseTypeChecker { }
 
-  @TypeQualifiers({ Nullable.class, Raw.class, NonNull.class, PolyNull.class, PolyAll.class })
   @SupportedLintOptions({"flow", "cast", "cast:redundant"})
   public class NullnessChecker extends BaseTypeChecker { }
 \end{Verbatim}
 
-
 The checker class must indicate the annotations
 that make up the type hierarchy for this checker (including
-polymorphic qualifiers), either via a
-\refqualclass{framework/qual}{TypeQualifiers} annotation or by overriding
+polymorphic qualifiers), either by placing the annotations inside a subfolder called \<qual> located in the same directory as the checker, or by overriding
 the
 \refmethodterse{framework/type}{AnnotatedTypeFactory}{createSupportedTypeQualifiers}{--}
-method.  Each argument to \<@TypeQualifiers> or value returned by
-\<createSupportedTypeQualifiers> is a class literal for a type qualifier
-whose definition bears the
-\refqualclass{framework/qual}{TypeQualifier} meta-annotation.
+method. By default, the polymorphic qualifier \refqualclass{framework/qual}{PolyAll} is supported by a checker. It can be optionally excluded.
+
+There are four recommended ways to write an override implementation of \refmethodterse{framework/type}{AnnotatedTypeFactory}{createSupportedTypeQualifiers}{--}
+
+\begin{enumerate}
+\item using reflective lookup to add all qualifiers, and exclude \code{@PolyAll}. Example:
+
+\begin{Verbatim}
+  @Override
+  protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+      return loadTypeQualifiersFromQualDir(true, null);
+  }
+\end{Verbatim}
+
+\item using reflective lookup to add qualifiers, exclude \code{@PolyAll}, and add a manual
+list of qualifiers. Example:
+
+\begin{Verbatim}
+@Override
+protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+    return loadTypeQualifiersFromQualDir(true,
+        new HashSet<Class<? extends Annotation>>(
+            Arrays.asList(H1Top.class, H1S1.class, H1S2.class,
+                H1Bot.class, H2Top.class, H2S1.class, H2S2.class,
+                H2Bot.class, H1Poly.class, H2Poly.class)));
+}
+\end{Verbatim}
+
+The use of a manual list is necessary for any qualifiers that are not located in
+the \<qual> folder, or for qualifiers which do not or cannot have the
+\sunjavadocanno{java/lang/annotation/Target.html}{Target} meta-annotation with the value 
+\<ElementType.TYPE\_USE> and/or \<ElementType.TYPE\_PARAMETER> (e.g. qualifiers that 
```

Comment:
Add a comma after "e.g.", both for grammar and for correct spacing in the LaTeX output.